config/initializers/smtp.rb

```diff
@@ -1,6 +1,22 @@
 # frozen_string_literal: true
-ActionMailer::Base.smtp_settings = {
+require 'uri'
+
+smtp_url = URI.parse(ENV["SMTP_URL"] || "smtp://localhost")
+smtp_url.port ||= 25
+smtp_url.user = ENV["SMTP_USER"] if ENV["SMTP_USER"]
+smtp_url.password = ENV["SMTP_PASSWORD"] if ENV["SMTP_PASSWORD"]
+
+smtp_settings = {
+  port:                 smtp_url.port,
+  address:              smtp_url.host,
   authentication:       'plain',
   enable_starttls_auto: false,
   openssl_verify_mode:  'none'
 }
+
+if smtp_url.user && smtp_url.password
```

Comment:
I added way more `if`s than I'd have liked, mostly because I don't want to make any assumptions on how ActionMailer will treat `nil` credentials.

Reply:
nil works jus fine, tried it just now ... so ifs can go away!

Reply:
and can directly assign then too